mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -230,6 +231,44 @@ private void resolve(ComponentModel model, Object owner) {
     return result.get();
   }
 
+  /**
+   * @param extensionModel the model which owns the {@code component}
+   * @param component a component
+   * @return The config (if any) that the {@code component} needs in order to function
+   */
+  public static Optional<ConfigurationModel> getConfigurationFromComponent(ExtensionModel extensionModel, NamedObject component) {
+    if (!(component instanceof ComponentModel)) {
```

Comment:
why do you need this? It seems you can just receive a ComponentModel instead of a NamedObject.